File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/state/orderstate/OrderInProgressState.java
// OrderInProgressState.java

package com.bigboxcode.designpattern.state.orderstate;

public class OrderInProgressState extends OrderState {
    public OrderInProgressState(OrderContext context) {
        super(context);
    }

    @Override
    public void process() {
        // Write code to process the order
        System.out.println("Processing the order");

        context.setState(context.getOrderDeliveredState());
    }
}

File: design-patterns/php/state/order/OrderInProgressState.php
<?php
// OrderInProgressState.php

namespace BigBoxCode\DesignPattern\State\Order;


class OrderInProgressState extends OrderState {
    public function __construct(OrderContext $context) {
        parent::__construct($context);
    }

    public function process(): void {
        // Write code to process the order
        echo "Processing the order\n";

        $this->context->setState($this->context->getOrderDeliverState());
    }
}
